store/src/main/java/org/apache/rocketmq/store/StreamMessageStore.java

```diff
@@ -0,0 +1,2573 @@
+                    maxPullSize = MAX_PULL_MSG_SIZE;
+                }
+                status = GetMessageStatus.NO_MATCHED_MESSAGE;
+                long maxPhyOffsetPulling = 0;
+                int cqFileNum = 0;
+
+                while (getResult.getBufferTotalSize() <= 0
+                        && nextBeginOffset < maxOffset
+                        && cqFileNum++ < this.messageStoreConfig.getTravelCqFileNumWhenGetMessage()) {
+                    ReferredIterator<CqUnit> bufferConsumeQueue = consumeQueue.iterateFrom(nextBeginOffset);
+
+                    if (bufferConsumeQueue == null) {
+                        status = GetMessageStatus.OFFSET_FOUND_NULL;
+                        nextBeginOffset = nextOffsetCorrection(nextBeginOffset, this.consumeQueueStore.rollNextFile(consumeQueue, nextBeginOffset));
+                        log.warn("consumer request topic: " + topic + "offset: " + offset + " minOffset: " + minOffset + " maxOffset: "
+                                + maxOffset + ", but access logic queue failed. Correct nextBeginOffset to " + nextBeginOffset);
+                        break;
+                    }
+
+                    try {
+                        long nextPhyFileStartOffset = Long.MIN_VALUE;
+                        while (bufferConsumeQueue.hasNext()
+                                && nextBeginOffset < maxOffset) {
+                            CqUnit cqUnit = bufferConsumeQueue.next();
+                            long offsetPy = cqUnit.getPos();
+                            int sizePy = cqUnit.getSize();
+
+                            boolean isInDisk = checkInDiskByCommitOffset(offsetPy, maxOffsetPy);
+
+                            if (cqUnit.getQueueOffset() - offset > maxFilterMessageCount) {
+                                break;
+                            }
+
+                            if (this.isTheBatchFull(sizePy, cqUnit.getBatchNum(), maxMsgNums, maxPullSize, getResult.getBufferTotalSize(), getResult.getMessageCount(),
+                                    isInDisk)) {
+                                break;
+                            }
+
+                            if (getResult.getBufferTotalSize() >= maxPullSize) {
+                                break;
+                            }
+
+                            maxPhyOffsetPulling = offsetPy;
+
+                            //Be careful, here should before the isTheBatchFull
+                            nextBeginOffset = cqUnit.getQueueOffset() + cqUnit.getBatchNum();
+
+                            if (nextPhyFileStartOffset != Long.MIN_VALUE) {
+                                if (offsetPy < nextPhyFileStartOffset) {
+                                    continue;
+                                }
+                            }
+
+                            if (messageFilter != null
+                                    && !messageFilter.isMatchedByConsumeQueue(cqUnit.getValidTagsCodeAsLong(), cqUnit.getCqExtUnit())) {
```

Comment:
For batch messages, if the consumer uses tags to consume, will the messages not be consumed?